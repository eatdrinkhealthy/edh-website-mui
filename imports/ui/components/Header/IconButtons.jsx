import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Icon from "@material-ui/core/Icon";

const IconButtons = () => (
  <div className="icon-buttons">
    <a href="mailto:info@eatdrinkhealthy.co">
      <span className="fa-stack fa-lg">
        <i className="fa fa-circle fa-stack-2x" />
        <i className="fa fa-envelope fa-stack-1x fa-inverse" />
      </span>
    </a>
    <a
      href="https://www.linkedin.com/company/16173480/"
      target="_blank"
      rel="noopener noreferrer"
    >
      <span className="fa-stack fa-lg">
        <i className="fa fa-circle fa-stack-2x" />
        <i className="fab fa-linkedin fa-stack-1x fa-inverse" />
      </span>
    </a>
  </div>
);

export default IconButtons;
