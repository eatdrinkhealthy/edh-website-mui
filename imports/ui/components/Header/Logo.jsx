import React from "react";

const Logo = () => (
  <div className="header-container header-left">
    <a href="/">
      <img src="images/logo.png" alt="Company Logo" />
    </a>
  </div>
);

export default Logo;
