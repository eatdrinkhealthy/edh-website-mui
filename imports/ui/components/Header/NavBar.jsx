import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";

const pathAndHash = (match, location, toHash) => {
  if (!match) {
    return false;
  }

  return match.url === location.pathname && toHash === location.hash;
};

const styles = theme => ({
  container: {
    // margin: "5px 0 5px 0",
  },
  selected: {
    color: theme.palette.primary.main,
    "&:hover": {
      color: theme.palette.primary.main,
    },
  },
});

// TODO: to set an anchor link active, could check location.hash
// if equal, set Contact to active
// Note, may need to use Link instead of NavLink to do so
const NavBar = ({ classes: { container, selected } }) => (
  <nav className={container}>
    <ul>
      <li>
        <NavLink
          exact
          to="/"
          activeClassName={selected}
          isActive={(m, l) => pathAndHash(m, l, "")}
        >
          HOME
        </NavLink>
      </li>
      <li>
        <NavLink
          exact
          to={{ pathname: "/", hash: "#contact" }}
          activeClassName={selected}
          isActive={(m, l) => pathAndHash(m, l, "#contact")}
        >
          CONTACT
        </NavLink>
      </li>
      <li>
        <NavLink
          exact
          to="/privacy-policy"
          activeClassName={selected}
          isActive={(m, l) => pathAndHash(m, l, "")}
        >
          PRIVACY
        </NavLink>
      </li>
    </ul>
  </nav>
);

NavBar.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default withStyles(styles)(NavBar);
