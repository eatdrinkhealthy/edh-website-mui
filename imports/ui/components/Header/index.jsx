/* eslint-disable arrow-parens, object-curly-newline */
import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Logo from "./Logo";
import NavBar from "./NavBar";
import IconButtons from "./IconButtons";

const styles = theme => ({
  root: {
    flexGrow: 1,
  },
  appBar: {
    borderBottom: `8px solid ${theme.palette.primary.main}`,
    boxShadow: "none",
  },
  toolBar: {
    backgroundColor: "white",
  },
});

// App component - represents the whole app
const Header = props => {
  const { classes } = props;

  return (
    <Fragment>
      <div className={classes.root}>
        <AppBar position="static" className={classes.appBar}>
          <Toolbar className={classes.toolBar}>
            <Logo />
            <NavBar />
            <IconButtons />
          </Toolbar>
        </AppBar>
      </div>
    </Fragment>
  );
};

Header.propTypes = {
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default withStyles(styles)(Header);
