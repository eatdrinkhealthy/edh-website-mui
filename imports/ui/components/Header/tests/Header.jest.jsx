/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Header from "../index";

describe("<Header />", () => {
  it("should match render snapshot", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
