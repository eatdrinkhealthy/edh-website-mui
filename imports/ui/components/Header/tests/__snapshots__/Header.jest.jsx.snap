// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match render snapshot 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-11 Header-appBar-2"
  >
    <div
      className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41 Header-toolBar-3"
    >
      <div
        className="Logo-container-44"
      >
        <a
          href="/"
        >
          <img
            alt="Company Logo"
            src="images/logo.png"
          />
        </a>
      </div>
    </div>
  </header>
</div>
`;
